integration-tests/src/env/setup.rs

```diff
@@ -618,16 +618,18 @@ fn process_peer_manager_message_default(
 
             hash_to_height.write().insert(*block.header().hash(), block.header().height());
         }
-        NetworkRequests::OptimisticBlock { optimistic_block } => {
+        NetworkRequests::OptimisticBlock { block_producers, optimistic_block } => {
             // TODO(#10584): maybe go through an adapter to facilitate testing.
-            for actor_handles in connectors {
-                actor_handles.client_actor.do_send(
-                    OptimisticBlockMessage {
-                        optimistic_block: optimistic_block.clone(),
-                        from_peer: PeerInfo::random().id,
-                    }
-                    .with_span_context(),
-                );
+            for (i, name) in validators.iter().enumerate() {
+                if block_producers.contains(name) {
+                    connectors[i].client_actor.do_send(
+                        OptimisticBlockMessage {
+                            optimistic_block: optimistic_block.clone(),
+                            from_peer: PeerInfo::random().id,
+                        }
+                        .with_span_context(),
+                    );
+                }
```

Comment:
mini nit1: invert the condition to continue early and reduce nesting
mini nit2: move the message to a temporary variable to not clump too many things into a single line

both optional and based on personal preference ;)